### PR TITLE
Add `@edgedb/ai` package for `ext::ai`

### PR DESCRIPTION
**Pull request by scotttrinh**
Provides a high-level library for use with the `ext::ai` extension.

### CODE REVIEW

File: packages/ai/src/core.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    public readonly client: Client,
+    options: Partial<AIOptions> = {},
+    context: Partial<QueryContext> = {}
+  ) {
+    this.baseUrl = EdgeDBAI.getBaseUrl(client);
+    this.options = {
+      model: options.model ?? "gpt-3.5-turbo",
+      prompt: options.prompt ?? {
+        name: "builtin::rag-default",
+      },
```
Comment (jaclarke) [inline packages/ai/src/core.ts:27]: I'm not sure we need these defaults. The prompt is optional anyway, as the server will pick a default prompt. And the model is dependent on which providers you have configured, so it doesn't seem safe to use an openai model as the default?

File: packages/ai/debug.mts
Comment (jaclarke) [inline packages/ai/debug.mts:1]: Maybe this should be moved to an example in a readme?

File: packages/ai/src/core.ts (hunk 1 above)
Comment (scotttrinh) [reply packages/ai/src/core.ts:27]: Oh, that's a good point. Will fix!

File: packages/ai/debug.mts
Comment (scotttrinh) [reply packages/ai/debug.mts:1]: Yeah, I can do that with this and the stream one. We should have full examples with servers in the example repo.

**Review by jaclarke — APPROVED** [review]

**Pull request merged**